### PR TITLE
Allowed "force enabling" of flat tables use for a product collection

### PR DESCRIPTION
**Pull request by loekvangool**
Fixes https://github.com/OpenMage/magento-lts/issues/2914

### CODE REVIEW

**Comment by loekvangool** [comment]
Example on how to implement a fast flat table collection while in an area that until now blocks access to flat tables:

```php
$productFlatHelper = Mage::helper('catalog/product_flat');
var_dump("<br>\$productFlatHelper->isFlatCollectionDisabled(): ".$productFlatHelper->isFlatCollectionDisabled()); // true if in cart
$productFlatHelper->enableFlatCollection(true); // call to the new method
var_dump("<br>\$productFlatHelper->isFlatCollectionDisabled(): ".$productFlatHelper->isFlatCollectionDisabled()); // false

// Get some products
$_products = Mage::getModel("catalog/product")->setPageSize(12)->setCurPage(1);
var_dump($collection->getSelect()->__toString()); // uses flat table

$productFlatHelper->resetFlatCollection();
var_dump("<br>\$productFlatHelper->isFlatCollectionDisabled(): ".$productFlatHelper->isFlatCollectionDisabled()); // false
```

**Comment by elidrissidev** [comment]
Can you add a space after the `if` to make PHP-CS-Fixer check pass.

File: app/code/core/Mage/Catalog/Helper/Product/Flat.php [hunk 1]
```diff
@@ -164,6 +164,20 @@ public function isAddChildData()
         return (int) Mage::getConfig()->getNode(self::XML_NODE_ADD_CHILD_DATA);
     }
 
+    /**
+     * Enable Catalog Product Flat
```
Comment (fballiano) [inline app/code/core/Mage/Catalog/Helper/Product/Flat.php:168]:
```suggestion
     * Enable Catalog Product Flat.
```

File: app/code/core/Mage/Catalog/Helper/Product/Flat.php [hunk 2]
```diff
@@ -164,6 +164,20 @@ public function isAddChildData()
         return (int) Mage::getConfig()->getNode(self::XML_NODE_ADD_CHILD_DATA);
     }
 
+    /**
+     * Enable Catalog Product Flat.
```
Comment (fballiano) [inline app/code/core/Mage/Catalog/Helper/Product/Flat.php:168]:
```suggestion
     * Enable Catalog Product Flat
```

**Review by fballiano — APPROVED** [review]
I do not love this PR because of https://github.com/OpenMage/magento-lts/issues/2914#issuecomment-1382710162 but, at the end of the day, it has to be forces by somebody in his/her own code so it's fine with me

**Review by ADDISON74 — APPROVED** [review]

**Pull request merged**